include/exec/task.hpp

```diff
@@ -306,18 +322,8 @@ namespace exec {
           return __task_awaitable<>{std::exchange(__self.__coro_, {})};
         }
 
-        // Specify basic_task's completion signatures
-        //   This is only necessary when basic_task is not generally awaitable
-        //   owing to constraints imposed by its _Context parameter.
-        template <class... _Ts>
-          using __task_traits_t =
-            std::execution::completion_signatures<
-              std::execution::set_value_t(_Ts...),
-              std::execution::set_error_t(std::exception_ptr),
-              std::execution::set_stopped_t()>;
-
         friend auto tag_invoke(std::execution::get_completion_signatures_t, const basic_task&, auto)
-          -> std::conditional_t<std::is_void_v<_Ty>, __task_traits_t<>, __task_traits_t<_Ty>>;
```

Comment:
I'm curious why you had to change the return type here.

Reply:
I tried to add the details to the commit message of the commit where I did this.

The original failure on GCC was:
```
task.hpp:370:17: error: invalid parameter type ‘void’
  370 |     friend auto tag_invoke(std::execution::get_completion_signatures_t,
      |                 ^~~~~~~~~~
task.hpp:370:17: error: in declaration ‘std::conditional_t<is_void_v<_Ty>, stdexec::completion_signatures<stdexec::__receivers::set_value_t(), stdexec::__receivers::set_error_t(std::__exception_ptr::exception_ptr), stdexec::__receivers::set_stopped_t()>, stdexec::completion_signatures<stdexec::__receivers::set_value_t(_Ty), stdexec::__receivers::set_error_t(std::__exception_ptr::exception_ptr), stdexec::__receivers::set_stopped_t()> > exec::__task::tag_invoke(stdexec::__get_completion_signatures::get_completion_signatures_t, const basic_task<_Ty, _Context>&, auto:82)’
```

GCC is unhappy about the evaluation of `__task_traits_t<void>` even though it is inside the `std::conditional_t`.  I promoted the return type to a template specialization so that the code never attempts to evaluate `__task_traits_t<void>` and only does `__task_traits_t<>` (when _Ty is void).